src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java

```diff
@@ -66,22 +66,28 @@ else if ("private".equals(token.getText())) {
      */
     public static Scope getSurroundingScope(DetailAST node) {
         Scope returnValue = null;
-        for (DetailAST token = node.getParent();
-             token != null;
-             token = token.getParent()) {
-            final int type = token.getType();
-            if (TokenUtil.isTypeDeclaration(type)) {
-                final DetailAST mods =
-                    token.findFirstToken(TokenTypes.MODIFIERS);
-                final Scope modScope = getScopeFromMods(mods);
-                if (returnValue == null || returnValue.isIn(modScope)) {
-                    returnValue = modScope;
+        if (TokenUtil.isOfType(node.getParent(), TokenTypes.OBJBLOCK)
+                && TokenUtil.isOfType(node.getParent().getParent(), TokenTypes.INTERFACE_DEF)) {
+            returnValue = Scope.PUBLIC;
```

Comment:
So when its in an interface, it's always public? We don't even have to check the modifiers to determine if that is the case?

I believe a newer version of Java allows private or some non-public in the interfaces.

Reply:
This assumption was made from [Java Language Specification JLS 9.3](https://docs.oracle.com/javase/specs/jls/se16/html/jls-9.html#jls-9.3) where the line states: **"Every field declaration in the body of an interface delaration is implicitly public, static, and final. It is permitted to redundantly specify any or all of these modifiers for such fields. "** 

I had found some articles related to private modifier. I will update you once I find JLS specifications for it. Until then, kindly wait for some time.

Reply:
Thanks, let me know when you are ready for me to look at it again.

Reply:
Sir, I have done changes, also I have checked the reports, they look fine two me. DesignForExtensionCheck and
InterfaceMemberImpliedModifierCheck are checks that might be affected, but are not affected by the already present test, new test according to the regression will be added as I do more tests.
Kindly check already present reports till then.

Reply:
@rnveach Changes are done. Please review at your convenience.

Reply:
I also don't understand how this solves the problem. In this code
```java
public interface Iface {
    class NoModsClass {
        public void test() { // expected violation
        }
    }
}
```
The class `Iface.NoModsClass` is implicitly public. There should be a violation, but `getSurroundingScope` returns `package` for method `test()`.